src/CMakeLists.txt

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+#-----------------------------------------------------------------------
+# Create core target
+
+# Note: static vs. shared is auto-determined by CMake's built-in
+# BUILD_SHARED_LIBS option.
+# TODO(DF): Allow customising library name (e.g. suffix)? See OCIO.
+add_library(openassetio)
+# Set good default target options.
+include(DefaultTargetProperties)
+set_default_target_properties(openassetio)
+
+
+#-----------------------------------------------------------------------
+# Target dependencies
+
+target_sources(openassetio
+    PRIVATE
+    openassetio/managerAPI/ManagerInterface.cpp)
+
+target_link_libraries(openassetio PUBLIC headers)
+
+
+#-----------------------------------------------------------------------
+# API export header
+
+# Definition for API header, to use for versioned namespacing.
+# TODO(DF): It may turn out this should go in a separate header. Also
+#   other projects have much more elaborate version headers. See OCIO
+#   (bundles version in ABI header) and OTIO (uses separate header),
+#   both of which include a long list of additional #defines.
+set(define_version
+    "#define OPENASSETIO_VERSION v${PROJECT_VERSION_MAJOR}_${PROJECT_VERSION_MINOR}")
+
+# TODO(DF): Allow customising namespace? See OCIO.
+
+# Use CMake utility to generate the export header.
+include(GenerateExportHeader)
+generate_export_header(
+    openassetio
+    # TODO(DF): What do we want to call this?
+    #   OIIO calls it export.h
+    #   OCIO calls it OpenColorABI.h
+    #   OTIO apprently doesn't hide symbols except for Python builds
+    EXPORT_FILE_NAME openassetio/abi.hpp
+    # #define that when found will configure as a static lib.
+    STATIC_DEFINE ${PROJECT_NAME}_STATIC_DEFINE
+    CUSTOM_CONTENT_FROM_VARIABLE define_version
+)
+
+target_include_directories(openassetio
+    PRIVATE
+    # For ABI header.
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>")
+
+install(
+    FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/openassetio/abi.hpp
+    DESTINATION
+    ${CMAKE_INSTALL_INCLUDEDIR}/openassetio/
+)
+
+#-----------------------------------------------------------------------
+# Install rules
+
+# Add to the set of installable targets.
+install(TARGETS openassetio EXPORT ${PROJECT_NAME}_EXPORTED_TARGETS)
+
+# TODO(DF): Generate and install .pdb on Windows. See OCIO.
```

Comment:
No blank line at EOF?

Reply:
Added blank lines and updated `.editorconfig` in f81c11d